cmd/main.go

```diff
@@ -34,6 +35,15 @@ import (
 	//+kubebuilder:scaffold:imports
 )
 
+// healthHandler checks the health of the server and returns a status code accordingly
+func healthHandler(w http.ResponseWriter, r *http.Request) {
+	if controller.ServerIsHealthy() {
+		w.WriteHeader(http.StatusOK)
+	} else {
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+}
+
```

Comment:
It would be better to inject the check function instead of referring to it statically, so you could unittest this.
```suggestion
// newHealthHandler returns an HTTP HandlerFunc that checks the health of the server by calling the given function and returns a status code accordingly
func newHealthHandler(checkFunc func() bool) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		if checkFunc() {
			w.WriteHeader(http.StatusOK)
		} else {
			w.WriteHeader(http.StatusInternalServerError)
		}
	}
}

```